src/client/dfs/dfs.c

```diff
@@ -1153,39 +1147,20 @@ dfs_mount(daos_handle_t poh, daos_handle_t coh, int flags, dfs_t **_dfs)
 	if (rc != 0)
 		D_GOTO(err_dfs, rc = daos_der2errno(rc));
 
-	prop = daos_prop_alloc(0);
-	if (prop == NULL) {
-		D_ERROR("Failed to allocate prop.");
-		D_GOTO(err_dfs, rc = ENOMEM);
-	}
-
-	rc = daos_pool_query(poh, NULL, &pool_info, prop, NULL);
-	if (rc) {
-		D_ERROR("daos_pool_query() Failed (%d)\n", rc);
-		daos_prop_free(prop);
-		D_GOTO(err_dfs, rc = daos_der2errno(rc));
-	}
-
 	/* Convert the owner information to uid/gid */
-	entry = daos_prop_entry_get(prop, DAOS_PROP_PO_OWNER);
+	entry = daos_prop_entry_get(prop, DAOS_PROP_CO_OWNER);
 	D_ASSERT(entry != NULL);
 	rc = daos_acl_principal_to_uid(entry->dpe_str, &dfs->uid);
-	if (rc != 0) {
-		D_ERROR("Unable to convert owner to uid\n");
-		daos_prop_free(prop);
-		D_GOTO(err_dfs, rc = daos_der2errno(rc));
-	}
+	if (rc != 0)
```

Comment:
A few different values can be returned by these conversion functions. Maybe we should check specifically for -DER_NONEXIST and -DER_INVAL. -DER_NONEXIST is exactly what it sounds like, the name was well-formed but not found. In this case -DER_INVAL would indicate a malformed principal string was entered (could mean that the property value got corrupted, for example). Both the uid and gid versions of this function can also return -DER_NOMEM, which seems like it would be worth failing the mount over.

Reply:
sure will fix that